tests/phpunit/tests/html-api/wpHtmlTagProcessor.php

```diff
@@ -51,6 +51,54 @@ public function test_get_tag_returns_open_tag_name() {
 		$this->assertSame( 'DIV', $p->get_tag(), 'Accessing an existing tag name did not return "div"' );
 	}
 
+	/**
+	 * @ticket 58009
+	 *
+	 * @covers WP_HTML_Tag_Processor::has_self_closing_flag()
+	 *
+	 * @dataProvider data_has_self_closing_flag
+	 *
+	 * @param string $html input HTML whose first tag might contain the self-closing flag `/`.
+	 * @param bool $flag_is_set whether the input HTML's first tag contains the self-closing flag.
+	 */
+	public function test_has_self_closing_flag_matches_input_html( $html, $flag_is_set ) {
+		$p = new WP_HTML_Tag_Processor( $html );
+		$p->next_tag( array( 'tag_closers' => 'visit' ) );
+
+		if ( $flag_is_set ) {
+			$this->assertTrue( $p->has_self_closing_flag(), 'Did not find the self-closing tag when it was present.' );
+		} else {
+			$this->assertFalse( $p->has_self_closing_flag(), 'Found the self-closing tag when it was absent.' );
+		}
+	}
+
+	/**
+	 * Data provider. HTML tags which might have a self-closing flag, and an indicator if they do.
+	 *
+	 * @return array[]
+	 */
+	public function data_has_self_closing_flag() {
+		return array(
+			// These should not have a self-closer, and will leave an element un-closed if it's assumed they are self-closing.
+			'Self-closing flag on non-void HTML element' => array( '<div />', true ),
+			'No self-closing flag on non-void HTML element' => array( '<div>', false ),
```

Comment:
It's interesting the linter doesn't complain about a misaligned arrow 🤔  I hate to be that person, but all the arrows need to be at the same offset.

Reply:
then don't be that person 😄 

I'll try this out, but I think the linter complains here if we line up the arrows. it has some line-length threshold at which it separates the alignment of shorter lines and longer lines.

Reply:
confirmed: the alignment that looks right is rejected because it doesn't follow the specific linting rules. now go away and take your legibility with you! 🙃